tests/unit/build_tests/test_io_map.py

```diff
@@ -260,6 +262,26 @@ def test_dynamic_container_composition(self):
         with self.assertRaises(TypeError):
             Baz1('My Baz', [1, 2, 3, 4], 'string attribute', 1000, attr3=98.6, attr4=1.0, baz2=bar)
 
+    def test_dynamic_container_composition_wrong_order(self):
+        baz_spec2 = GroupSpec('A composition inside', data_type_def='Baz2',
+                              data_type_inc=self.bar_spec,
+                              attributes=[
+                                  AttributeSpec('attr3', 'an example float attribute', 'float'),
+                                  AttributeSpec('attr4', 'another example float attribute', 'float')])
+
+        baz_spec1 = GroupSpec('A composition test outside', data_type_def='Baz1', data_type_inc=self.bar_spec,
+                              attributes=[AttributeSpec('attr3', 'an example float attribute', 'float'),
+                                          AttributeSpec('attr4', 'another example float attribute', 'float')],
+                              groups=[GroupSpec('A composition inside', data_type_inc='Baz2')])
+        self.spec_catalog.register_spec(baz_spec1, 'extension.yaml')
+        self.spec_catalog.register_spec(baz_spec2, 'extension.yaml')
+
+        # Setup all the data we need
+        msg = ("Cannot dynamically generate class for type 'Baz1'. Type 'Baz2' does not exist. "
```

Comment:
Should get_class recurse through and define Baz2 when it encounters it?

@bendichter what do you think?

Reply:
Yeah that seems like a good idea. I think this would leave us open to the possibility of an infinite loop if baz1 is in baz2 and baz2 is in baz1. Should we protect against an infinite loop, or do we want to let python handle that?

Reply:
I gave this a shot, but recursing through the hierarchy of types included in the current type is quite complicated. Especially if they are in different namespaces. I'm going to punt this into it's own issue so we can get this change out.